src/main/Production/ProductionController.java

```diff
@@ -24,66 +27,86 @@ public ProductionController(OrgDao orgDao, UserDao userDao) {
   public Handler createOrg =
       ctx -> {
         Organization organization = ctx.bodyAsClass(Organization.class);
+
         organization.setId(new ObjectId());
+        organization.setCreationDate(new Date());
+
         orgDao.save(organization);
       };
 
   public Handler readOrg =
       ctx -> {
         ObjectId objectId = new ObjectId(ctx.pathParam("orgId"));
         Optional<Organization> organizationOptional = orgDao.get(objectId);
-        organizationOptional.ifPresent(organization -> ctx.result(organization.toString()));
+        organizationOptional.ifPresent(organization -> ctx.json(organization.serialize().toMap()));
       };
 
   public Handler updateOrg =
       ctx -> {
-        Organization organization = ctx.bodyAsClass(Organization.class);
-        orgDao.update(organization);
+        ObjectId objectId = new ObjectId(ctx.pathParam("orgId"));
+        var updateRequest = ctx.bodyAsClass(OrganizationUpdateRequest.class);
+
+        Optional<Organization> organizationOptional = orgDao.get(objectId);
+        organizationOptional.ifPresent(value -> {
+          var organization = value.updateProperties(updateRequest);
+          orgDao.update(organization);
+          ctx.json(organization.serialize().toMap());
+        });
```

Comment:
I'd love feedback on this new entity update pattern, which essentially uses a new class as a means of whitelisting the properties that can be updated

Reply:
I think it would make sense to use this same pattern for entity creation as well, if there aren't any objections.

Reply:
@nrod80 does this update entity pattern have things like validation built in? I like the idea of the update request class

Reply:
@crchong1 There isn't any sort of request validation built in, but it should be pretty straightforward to add something here: https://github.com/keepid/keepid_server/pull/110/files#diff-5c5d7596603b526e05330bc5ec55fe98d601a310f184cf827191a98f992e1cabR327-R365

Reply:
nice :)